src/main/java/lotto/domain/WinningStatistics.java

```diff
@@ -1,37 +1,28 @@
 package lotto.domain;
 
-import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
 public class WinningStatistics {
-    private final Map<RANK, Integer> winningStatistics;
+    private final Map<Rank, Long> winningStatistics;
 
-    public WinningStatistics() {
-        winningStatistics = new HashMap<>();
+    public WinningStatistics(Lottos lottos, WinningLotto winningLotto) {
+        List<Rank> lottoMatches = lottos.compareLottos(winningLotto);
 
-        winningStatistics.put(RANK.ZERO, 0);
-        winningStatistics.put(RANK.ONE, 0);
-        winningStatistics.put(RANK.TWO, 0);
-        winningStatistics.put(RANK.THREE, 0);
-        winningStatistics.put(RANK.FOUR, 0);
-        winningStatistics.put(RANK.FIVE, 0);
-        winningStatistics.put(RANK.SIX, 0);
-    }
+        Map<Rank, Long> rankMap = new HashMap<>();
 
-    public void setRankCount(List<RANK> rankList) {
-        for (RANK rank : rankList) {
-            addCount(rank);
+        for (Rank rank : lottoMatches) {
+            rankMap.put(rank, rankMap.getOrDefault(rank, 0L) + 1);
         }
+        winningStatistics = rankMap;
     }
 
-    private void addCount(RANK rank) {
-        int count = winningStatistics.get(rank);
-        winningStatistics.put(rank, count + 1);
+    public Map<Rank, Long> getWinningStatistics() {
+        return winningStatistics;
```

Comment:
```suggestion
        return Collections.unmodifiableMap(winningStatistics);
```
요런 친구들은 더더욱 불변으로 관리해주면 좋을 것 같아요 ^_^